addons/mail/models/mail_thread.py

```diff
@@ -1247,10 +1247,17 @@ def message_process(self, model, message, custom_values=None,
         if strip_attachments:
             msg_dict.pop('attachments', None)
 
-        existing_msg_ids = self.env['mail.message'].search([('message_id', '=', msg_dict['message_id'])], limit=1)
+        message_ids = [msg_dict['message_id']]
```

Comment:
just noticed now, but can't that make false positive when an Odoo database send an email to an other Odoo database ? (it was already the case before, so maybe that flow is never done in practice)